lib/active_merchant/billing/gateways/litle.rb

```diff
@@ -332,6 +381,10 @@ def add_address(doc, address)
       end
 
       def add_order_source(doc, payment_method, options)
+        # TODO(zmd): mutating options like this feels wrong... ponder the error
+        #   of my ways further before committing to this...
+        order_source_options_for_stored_credential(options)
+
         if options[:order_source]
```

Comment:
Yeah let's have this be something like `if options[:order_source] || parameters[:stored_credential] && !parameters[:stored_credential].values.all?(&:nil?)` and munge together the condititionals from your `order_source_options_for_stored_credential` with the below.

Reply:
Yah, I didn't like mutating the options hash to achieve the necessary result.

I think I've found a good solution, at least one that has the same desired effect (setting orderSource under certain conditions when stored_credential is being passed, but still ensuring logic falls back to default for CIT used cases, without mutating the options hash directly).

https://github.com/activemerchant/active_merchant/pull/3155/commits/2ccd672242a4743ae39f1c53fbdc6340337bfa51

Please let me know if my proposed solution misses something that you were seeing here in my original hack. Thanks!